libbeat/autodiscover/builder/helper.go

```diff
@@ -64,6 +66,19 @@ func getStringAsList(input string) []string {
 	return list
 }
 
+// GetHintAsConfig can read a hint in the form of a stringified JSON and return a common.MapStr
+func GetHintAsConfig(hints common.MapStr, key, config string) []common.MapStr {
+	if str := GetHintString(hints, key, config); str != "" {
+		cfg := []common.MapStr{}
```

Comment:
Would it make sense to accept both a list or a single config `{...}`? I guess in most cases the user wants just a config, with some special exceptions where you may want more